src/app/pages/mentor-search-directory/mentor-search-directory.page.ts

```diff
@@ -0,0 +1,209 @@
+  selectedChips: boolean;
+  urlFilterData: string;
+  setPaginatorToFirstpage: boolean;
+  page: any = 1;
+  data: any;
+  isLoaded: boolean;
+  // mentors = [];
+  totalCount: any;
+  limit: any;
+  flatChips = [];
+
+  constructor(
+    private router: Router,
+    private loaderService: LoaderService,
+    private httpService: HttpService,
+    private modalCtrl: ModalController,
+    private permissionService: PermissionService,
+    private formService: FormService
+  ) { }
+
+  async ngOnInit() {
+    this.getMentors();
+    this.permissionService.getPlatformConfig().then((config)=>{
+      this.overlayChips = config?.result?.search_config?.search?.mentor?.fields;
+    })
+    let data = await this.formService.filterList('profile')
+    this.filterData = this.transformData(data)
+  }
+
+  onSearch(){
+    this.getMentors()
+  }
+  
+  selectChip(chip) {
+    this.selectedChipLabel = chip.label;
+    this.selectedChipName = chip.name;
+    this.isOpen = false;
+    this.getMentors()
+  }
+
+  closeCriteriaChip(){
+    this.selectedChipLabel = "";
+    this.selectedChipName = "";
+    this.getMentors()
+  }
+
+  removeChip(chip: string,index: number) {
+    this.flatChips.splice(index, 1);
+    this.removeFilteredData(chip)
+    this.getFilteredData();
+    this.getMentors()
+  }
+
+  async onClickFilter() {
+    let modal = await this.modalCtrl.create({
+      component: FilterPopupComponent,
+      cssClass: 'filter-modal',
+      componentProps: { filterData: this.filterData }
+    });
+
+    modal.onDidDismiss().then(async (dataReturned) => {
+      this.filteredDatas = []
+      if (dataReturned !== null) {
+        if (dataReturned?.data?.data?.selectedFilters) {
+          for (let key in dataReturned?.data?.data?.selectedFilters) {
+            this.filteredDatas[key] = dataReturned?.data?.data?.selectedFilters[key].slice(0, dataReturned?.data?.data?.selectedFilters[key].length).map(obj => obj.value).join(',').toString()
+          }
+          this.selectedChips = true;
+        }
+        this.extractLabels(dataReturned?.data?.data?.selectedFilters);
+        this.getFilteredData();
+      }
+      this.page = 1;
+      this.setPaginatorToFirstpage = true;
+      this.getMentors()
+    });
+    modal.present();
+  }
+
+  transformData(responseData) {
+    const entityTypes = responseData.entity_types;
+  
+    const filterData = Object.keys(entityTypes).map(type => {
+      const entityType = entityTypes[type][0];
+      return {
+        title: entityType.label,
+        name: entityType.value,
+        options: entityType.entities.map(entity => ({
+          label: entity.label,
+          value: entity.value
+        })),
+        type: "checkbox"
+      };
+    });
+  
+    return filterData;
+  }
+
+  extractLabels(data) {
+    this.flatChips = [];
+    for (const key in data) {
+      if (data.hasOwnProperty(key)) {
+        this.flatChips.push(...data[key]);
+      }
+    }
+  }
+
+  getFilteredData() {
```

Comment:
can u change all the name to a meaning full name? For eg: is this function giving filterData? no right. its creating a query string i think.